src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs

```diff
@@ -46,16 +47,30 @@ public void CreateInstance_WhenMessageInitialized_ShouldBeThreadsafe()
             });
         }
 
+#if NETCOREAPP
+#nullable enable
         [Test]
         public void Should_handle_messages_with_nullable_reference_types()
         {
             var mapper = new MessageMapper();
 
             // Type defined in separate assembly as a workaround
             // because we can't use nullable refeference types yet
-            mapper.CreateInstance<WithDodgyNullable.IMyMessage>();
+            mapper.CreateInstance<IMessageWithNullableProperties>();
         }
 
+        public interface IMessageWithNullableProperties : ICommand, IMessage
+        {
+            string Schema { get; set; }
+
+            string? Notes { get; set; }
+            object[]? NullableComplexType { get; set; }
```

Comment:
This handles the first scenario mentioned in #5752, but what about https://github.com/Particular/NServiceBus/issues/5752#issuecomment-794065319?

Would be good to see that also in a test.

Reply:
Good catch. I added that, too.